servers/castor/edna.nix

```
@@ -0,0 +1,121 @@
+{pkgs, lib, config, ...}:
+  let
+    inherit (builtins) toString typeOf toJSON;
+    inherit (lib) collect concatStringsSep;
+    inherit (pkgs) writeText;
+
+    dns_names = with config.networking;
+      concatStringsSep "," [ "${hostName}.${domain}" "staging.edna.serokell.team" ];
+
+    profile = "/nix/var/nix/profiles/per-user/deploy/edna-docker";
+
+    ports = {
+      backend = {
+        api = 9000;
+      };
+      frontend = {
+        http = 80;
+        https = 443;
+      };
+      prometheus = {
+        node = 9100;
+      };
+    };
+
+    pginit = writeText "init.sql"
+    ''
+      CREATE DATABASE edna;
+    '';
+  in
+  {
+    networking.firewall.allowedTCPPorts = collect (a: (typeOf a) == "int") { inherit (ports) frontend prometheus; };
+
+    virtualisation.docker = {
+      enable = true;
+      logLevel = "warn";
+      storageDriver = "overlay2";
+      volumes = [ "pgdata" ];
+      networks.edna = {};
+    };
+
+    systemd.services.docker-frontend.serviceConfig.RestartSec = "5m";
+
+    virtualisation.oci-containers.containers = let
+      commonOptions = {
+        extraOptions = [
+          # Put all nodes on the same private network
+          "--network=edna"
+
+          # PostgreSQL during initdb
+          "--shm-size=256MB"
+        ];
+      };
+
+    in
+    {
+      frontend = {
+        image = "ghcr.io/serokell/edna-backend";
```

```suggestion
        image = "ghcr.io/serokell/edna-frontend";
```

Maybe I misunderstand something, but I guess it should be `frontend`.

---

No, you're right. This silly mistake had me running in circles for half a day. I had a fix locally, but forgot to push. Nice catch.